cli/cmd/wallet.go

```diff
@@ -208,8 +209,8 @@ var (
 			questions := []*survey.Question{
 				{
 					Name:     "data",
-					Prompt:   kind.Prompt(),
-					Validate: kind.DataValidator(),
+					Prompt:   kind.Prompt(wfCfg["algorithm"].(string)),
+					Validate: kind.DataValidator(wfCfg["algorithm"].(string)),
```

Comment:
Uh, this looks ugly, what will happen for wallets that don't have an `algorithm` config? How the config dictionary looks like is an internal detail of the wallet.

Reply:
I added new `DataPrompt()` and `DataValidator()` methods to the wallet factory for this.